sdk/eventhub/azure-eventhub/azure/eventhub/_buffered_producer/_buffered_producer_dispatcher.py

```diff
@@ -46,9 +45,15 @@ def __init__(
         self._partition_resolver = PartitionResolver(self._partition_ids)
         self._max_wait_time = max_wait_time
         self._max_buffer_length = max_buffer_length
-        self._max_concurrent_sends = max_concurrent_sends
-        self._existing_executor = bool(executor)
-        self._executor = executor or ThreadPoolExecutor(max_worker)
+        self._existing_executor = False
+
+        if not executor:
+            self._executor = ThreadPoolExecutor(max_worker)
+        elif isinstance(executor, ThreadPoolExecutor):
+            self._executor = executor
+        elif isinstance(executor, int):
+            self._existing_executor = True
```

Comment:
This should be false here, the only scenario where it would be true is line 52, i.e. the customer instantiated the ThreadPoolExecutor themselves. The reason for tracking this is so that we don't shutdown the executor if we didn't own it to start with (i.e. line 146)